src/lib/handleRef.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react'
+
+/** A type that ensures that passed prop is defined in props and matches the type. */
+type HandleRefProps<TProp extends string, N> = { [K in TProp]?: React.Ref<N> }
+
+/**
+ * The function that correctly handles passing refs.
+ *
+ * @param props All components props
+ * @param propName The name of a ref prop
+ * @param node A node that should be passed by ref
+ */
+const handleRef = <P extends string, N>(props: HandleRefProps<P, N>, propName: P, node: N) => {
```

Comment:
maybe `setRefValue`/`setRefNode` name would be more expressive?

Reply:
For me, the current name is more obvious because we using the `handle*Ref` pattern for naming methods.

Reply:
`ref` can be a function or an object. When you're going to call a function you will never say `set function`, you will say `call function`.

Yesterday, we discussed the naming issue and agreed that `setRef`/`invokeRef`/`applyRef` doesn't provide more sense. I will ask Levi about this again.

Reply:
Let's merge it as is for now.  We've had discussions on this in the past and have decided that `this.ref` / `this.*Ref` should be used for instance property naming and that `handle*` should be used for instance methods than handle callbacks (events and refs).

https://github.com/Semantic-Org/Semantic-UI-React/issues/607
https://github.com/Semantic-Org/Semantic-UI-React/pull/1360#discussion_r102268814
https://github.com/Semantic-Org/Semantic-UI-React/pull/1360#issuecomment-281530696

I don't have time at present to find the discussions on `handle*` naming for callback handlers, but it also a community standard and something that has already been debated.

***

If there is still strong disagreement on this @kuzhelov let's please address it by an RFC and vote after this is merged.